types/shares_test.go

```diff
@@ -439,6 +438,26 @@ func Test_parseMsgShares(t *testing.T) {
 	}
 }
 
+func TestParsePaddedMsg(t *testing.T) {
```

Comment:
Do we have tests for non-interactive defaults somewhere? Testing all scenarios?

Reply:
I'll list my other comment here
> note that most of the testing is occurring in preexisting tests since this is mainly a refactor. We will also be testing this much more in the app when we actually start using namespaced padding in an upcoming PR

but this _should_ (famous last words) be testing the scenarios we need as there fortunately not that many for padding. The only thing this implementation needs to do is to ignore padding, and in a flattened set a of shares, this is simple since messages with a length delimiter of 0 **must** be padding.

the most important thing to test with this implementation is that it is equivalent with the old implementation and that it catches the edge cases of decoding a square. It appears to be using the existing tests. We have a test to crudely fuzz this (before go 1.18's official fuzzing) which simply runs `TestDataFromSquare` (which round trip tests random data) on a loop for some period of time. I ran it for a few minutes straight with no failing tests, which gives me confidence in that portion.

Reply:
Gonna further dig into this. Thanks a lot.